src/linse/typedsequence.py

```diff
@@ -160,26 +169,38 @@ class Word(TypedSequence):
     item_separator = ' + '
 
     @classmethod
-    def from_string(cls, s: str, **kw):
+    def from_string(cls, s: str, separator=None, **kw):
+        separator = separator or cls.item_separator
         kw['type'] = Morpheme
         # We assume s is a list of morphemes separated by +:
         return cls(iterable=[
-            Morpheme.from_string(m.strip()) for m in s.split(cls.item_separator.strip())], **kw)
+            Morpheme.from_string(m.strip()) for m in s.split(separator.strip())], **kw)
+
+    @classmethod
+    def from_segments(cls, s, separator=None, **kw):
+        separator = separator or cls.item_separator
+        pattern = r"\s+" + re.escape(separator.strip()) + r"\s+"
+        return cls(iterable=[
+            Morpheme.from_segments(m) for m in re.split(pattern, s)], **kw)
 
     def to_text(self):
         return ''.join(m.to_text() for m in self)
 
+    def reversed_segments(self):
+        return Word([m[::-1] for m in self[::-1]])
+
```

Comment:
Looks good to me.

What do we do with the other changes in this PR?

Reply:
That's up to you to decide, of course ;) but since I didn't touch any of the existing methods, I think it would be no harm to keep the `from_segments` method around -- or do you suggest doing something differently?